db/version_builder.cc

```diff
@@ -782,6 +782,19 @@ class VersionBuilder::Rep {
     return Status::OK();
   }
 
+  Status ApplyCompactCursors(int level,
+                             const InternalKey& smallest_uncompacted_key) {
+    if (level < 0 || level > num_levels_) {
```

Comment:
Maybe we should ignore the update when `level >= num_levels_`. I think this would introduce an error if the user reopened with a reduced num_levels even after clearing the files out of the levels undergoing removal.

Reply:
Resolving because this is a duplicate of https://github.com/facebook/rocksdb/pull/10107#discussion_r902052331